src/pages/UserGarden.js

```diff
@@ -56,15 +56,26 @@ export const UserGarden = () => {
     dispatch(user.actions.setError(null));
     dispatch(plants.actions.setItems([]));
     navigate('/')
+  };
+
+  // function that takes a new plant as a param and updates the state of the plants
+  // array using spread operator. it creates a new array with the existing plants
+  // and the new plant added to it.
+  const handleAddPlant = (newPlant) => {
+    dispatch(plants.actions.setItems([...plantItems, newPlant]));
```

Comment:
AMAZING! You're so smart!
I wonder if we need to do a similar thing when deleting a plant, and if that would stop all of the immer errors we're getting in this version

Reply:
I think we need the delete function to be a reduces for the plants slice so it will be slightly different